pkg/providers/cloudstack/validator.go

```diff
@@ -200,6 +200,9 @@ func (v *Validator) ValidateClusterMachineConfigs(ctx context.Context, cloudStac
 		if len(machineConfig.Spec.Template.Id) == 0 && len(machineConfig.Spec.Template.Name) == 0 {
 			return fmt.Errorf("template is not set for CloudStackMachineConfig %s. Default template is not supported in CloudStack, please provide a template name or ID", machineConfig.Name)
 		}
+		if err, fieldName, fieldValue := machineConfig.Spec.DiskOffering.Validate(); err != nil {
```

Comment:
this should be part of config manager validation

Reply:
There will be a follow-up PR to move static validation away from validator.go